protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobBatchRecord.java

```diff
@@ -127,4 +130,40 @@ public JobBatchRecord setTruncated(boolean truncated) {
   public boolean getTruncated() {
     return truncatedProp.getValue();
   }
+
+  @Override
+  public String getType() {
+    return BufferUtil.bufferAsString(typeProp.getValue());
+  }
+
+  @Override
+  public String getWorker() {
+    return BufferUtil.bufferAsString(workerProp.getValue());
+  }
+
+  @Override
+  public Duration getTimeoutDuration() {
+    return Duration.ofMillis(timeoutProp.getValue());
+  }
+
+  public int getMaxJobsToActivate() {
+    return maxJobsToActivateProp.getValue();
+  }
+
+  @Override
+  public List<Long> getJobKeys() {
+    return StreamSupport.stream(jobKeysProp.spliterator(), false)
+        .map(LongValue::getValue)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public List<JobRecordValue> getJobs() {
+    return StreamSupport.stream(jobsProp.spliterator(), false).collect(Collectors.toList());
```

Comment:
I'm not sure how this works but I assume this will not create a copy right? If this is the case then we will have problems here, because if we reference these jobs and read a new JobBatchRecord the referenced job will change since the buffer is reused by the buffers

Reply:
This creates a new list with the JobRecords in jobProp. Just to be sure, what you're saying is that if we reuse the batch record the referenced jobs will change? Because I'd assume that the JobRecords to which the returned list points wouldn't change